src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java

```diff
@@ -0,0 +1,119 @@
+import uk.gov.hmcts.reform.bulkscanprocessor.validation.OcrValidator;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Date;
+import java.util.Optional;
+
+import static java.util.Collections.singletonList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class BlobProcessorTaskTest {
+    private BlobProcessorTask blobProcessorTask;
+
+    @Mock
+    private BlobManager blobManager;
+
+    @Mock
+    private DocumentProcessor documentProcessor;
+
+    @Mock
+    private EnvelopeProcessor envelopeProcessor;
+
+    @Mock
+    private EnvelopeRepository envelopeRepository;
+
+    @Mock
+    private ProcessEventRepository eventRepository;
+
+    @Mock
+    private ContainerMappings containerMappings;
+
+    @Mock
+    private OcrValidator ocrValidator;
+
+    @Mock
+    private ServiceBusHelper notificationsQueueHelper;
+
+    @Mock
+    private CloudBlobContainer container;
+
+    @Mock
+    private CloudBlockBlob cloudBlockBlob;
+
+    @Mock
+    private ListBlobItem blob;
+
+    @Mock
+    private BlobProperties blobProperties;
+
+    @Mock
+    private Date date;
+
+    @Mock
+    private BlobInputStream blobInputStream;
+
+    @BeforeEach
+    void setUp() {
+        blobProcessorTask = new BlobProcessorTask(
+            blobManager,
+            documentProcessor,
+            envelopeProcessor,
+            envelopeRepository,
+            eventRepository,
+            containerMappings,
+            ocrValidator,
+            notificationsQueueHelper,
+            false
+        );
+    }
+
+    @Test
+    void processBlobs_should_not_call_envelopeProcessor_if_failed_to_load_file() throws Exception {
+        // given
+        given(blobManager.listInputContainers()).willReturn(singletonList(container));
+        given(container.listBlobs()).willReturn(singletonList(blob));
+        given(blob.getUri()).willReturn(URI.create("file.zip"));
+        given(container.getBlockBlobReference("file.zip")).willReturn(cloudBlockBlob);
+        given(container.getName()).willReturn("cont");
+        given(envelopeProcessor.getEnvelopeByFileAndContainer("cont", "file.zip"))
+            .willReturn(null);
+        given(cloudBlockBlob.exists()).willReturn(true);
+        given(cloudBlockBlob.getProperties()).willReturn(blobProperties);
+        given(blobProperties.getLastModified()).willReturn(date);
+        given(date.before(any(Date.class))).willReturn(true);
+        given(blobManager.acquireLease(any(CloudBlockBlob.class), anyString(), anyString()))
+            .willReturn(Optional.of("lease"));
+        given(cloudBlockBlob.openInputStream()).willReturn(blobInputStream);
+        given(blobInputStream.read(any())).willThrow(new IOException());
+
+        // when
+        blobProcessorTask.processBlobs();
```

Comment:
will this method not throw ZipFileLoadException ?

Reply:
This exception won't be thrown. It will be caught and logged within this call.